### PR TITLE
Cleanup

### PR DESCRIPTION
**Pull request by gentlecat**
:sparkles: :sparkles: :sparkles:

cc @mayhem, @alastair

### CODE REVIEW

File: webserver/views/user.py [hunk 1]
```diff
@@ -14,43 +15,44 @@ def lastfmscraper(user_id):
...
-        user = db.user.get_by_mb_id(user_id)
-        if user is None:
-            raise NotFound("Can't find this user.")
 
     if lastfm_username:
-        params = {"base_url": url_for("user.lastfmscraper", user_id=user_id, _external=True),
-                "user_token": user.auth_token,
-                "lastfm_username": lastfm_username}
-        loader = render_template("user/loader.js", **params)
+        loader = render_template("user/loader.js", **{
```
Comment (alastair) [inline webserver/views/user.py]: if you're going to do this inline you may as well use kwargs

File: webserver/views/user.py [hunk 2]
```diff
@@ -14,43 +15,44 @@ def lastfmscraper(user_id):
...
+
+
+def _get_user(user_id):
+    if current_user.is_authenticated() and \
+       current_user.musicbrainz_id == user_id:
+        return current_user
+    else:
+        user = db.user.get_by_mb_id(user_id)
+        if user is None:
+            raise NotFound("Can't find this user.")
```
Comment (alastair) [inline webserver/views/user.py]: I'm not sure about returning view/flask-specific exceptions from helper methods. However, if this was a generic exception then we'd need a try/except in the view. What do you think?

File: webserver/views/user.py [hunk 3]
```diff
@@ -7,6 +7,7 @@
 
 user_bp = Blueprint("user", __name__)
 
+
 @user_bp.route("/lastfmscraper/<user_id>.js")
```
Comment (alastair) [inline webserver/views/user.py]: Not part of this commit, but we thought that this may not be a good place for this url, as it is in the `/user/` namespace (effectively preventing us having a user called `lastfmscraper`, however rare it may be)

File: webserver/views/api.py [hunk 4]
```diff
@@ -208,9 +96,132 @@ def get_listens(user_id):
...
+        messybrainz_id = messy_response['messybrainz_id']
+    except KeyError:
+        current_app.logger.error("MessyBrainz did not return a proper id")
+        raise InternalServerError
+
+    recording_id = messy_response.get('recording_id', None)
+    artist_id = messy_response.get('artist_id', None)
+    release_id = messy_response.get('release_id', None)
+
+    return messybrainz_id, recording_id, artist_id, release_id
```
Comment (alastair) [inline webserver/views/api.py]: The content in here may change in the future. It may be better to just return the `messy_response` directly

File: webserver/views/api.py [hunk 5]
```diff
@@ -20,140 +19,30 @@
...
 
-        if data['listen_type'] not in ('playing_now', 'single', 'import'):
-            log_and_raise_bad_request("JSON document requires a valid listen_type key", payload)
-
-        if (data['listen_type'] == "single" or data['listen_type'] == 'playing_now') and len(payload) > 1:
-            log_and_raise_bad_request("JSON document contains more than listen for a single/playing_now. "
-                             "It should contain only one. ", payload)
-    except KeyError:
-        log_and_raise_bad_request("Invalid JSON document submitted.", raw_data)
+    payload = _get_listen_payload(data, raw_data)
```
Comment (alastair) [inline webserver/views/api.py]: `get` or `validate`?

File: webserver/views/api.py [hunk 6]
```diff
@@ -20,140 +19,30 @@
...
 
     producer = SimpleProducer(_kafka)
```
Comment (alastair) [inline webserver/views/api.py]: can you move this into a kafla module (like we have database stuff in db). The module won't get very complex, so perhaps just a single file. How about `data/kafka.py` and `data/cassandra.py` ?

File: webserver/views/api.py [hunk 7]
```diff
@@ -208,9 +96,132 @@ def get_listens(user_id):
         del temp['user_id']
         listen_data.append(temp)
 
-    payload = {}
-    payload['user_id'] = user_id
-    payload['count'] = count
-    payload['listens'] = listen_data
+    return json.dumps({'payload': {
```
Comment (alastair) [inline webserver/views/api.py]: is this an api endpoint? is it better to use `jsonify` in order to send correct headers

File: webserver/views/user.py (hunk 1 above)
Comment (gentlecat) [inline webserver/views/user.py]: Yeah, that's how we do in other places too. Done.

File: webserver/views/user.py (hunk 2 above)
Comment (gentlecat) [inline webserver/views/user.py]: I think it's fine as long as this function is used within views in the same module.

File: webserver/views/user.py (hunk 3 above)
Comment (gentlecat) [inline webserver/views/user.py]: Yep, this view probably should be in a separate blueprint.

File: webserver/views/api.py (hunk 5 above)
Comment (gentlecat) [inline webserver/views/api.py]: It's both actually.

File: webserver/views/api.py (hunk 7 above)
Comment (gentlecat) [inline webserver/views/api.py]: Yep.

File: webserver/views/api.py (hunk 6 above)
Comment (gentlecat) [inline webserver/views/api.py]: Can we do it in another PR?

Comment (alastair) [inline webserver/views/api.py]: no problem

**Comment by gentlecat** [comment]
:ship::anchor:?

File: webserver/views/api.py (hunk 5 above)
Comment (alastair) [inline webserver/views/api.py]: i don't like the name of the method if it also validates

File: webserver/views/user.py (hunk 3 above)
Comment (alastair) [inline webserver/views/user.py]: ok, can you make a note somewhere? we have LB in jira, right?

File: webserver/views/user.py (hunk 2 above)
Comment (alastair) [inline webserver/views/user.py]: ok, fine

File: webserver/views/api.py (hunk 5 above)
Comment (gentlecat) [inline webserver/views/api.py]: I can extract validation part into `validate_listen_payload` and put getting where it was. Would that work?

**Comment by alastair** [comment]
yes :rocket:

File: webserver/views/api.py (hunk 5 above)
Comment (alastair) [inline webserver/views/api.py]: yeah, better
